src/main/scala/za/co/absa/hyperdrive/trigger/scheduler/utilities/Configs.scala

```diff
@@ -29,13 +29,14 @@ private object Configs {
 
   def getMapFromConf(propertyName: String): Map[String, String] = {
     Try {
-      val list = Configs.conf.getObjectList(propertyName).asScala
-      (for {
-        item: ConfigObject <- list
-        entry <- item.entrySet().asScala
-        key = entry.getKey
-        value = entry.getValue.unwrapped().toString
-      } yield (key, value)).toMap
+      def getKeys(path: String): Seq[String] = {
+        Try(Configs.conf.getObject(path).keySet().asScala).getOrElse(Set.empty[String]) match {
+          case keys if keys.nonEmpty => keys.flatMap(k => getKeys(s"$path.$k")).toSeq
+          case keys if keys.isEmpty => Seq(path)
+        }
+      }
+      val keys = getKeys(propertyName)
+      keys.map(k => (k.stripPrefix(s"$propertyName."), conf.getString(k))).toMap
     }.getOrElse(Map.empty[String, String])
```

Comment:
A test would be good for this non-trivial logic. Also, is it not possible to get all keys and then just filter them? It seems over-complicated to have recursion here. (Not your fault I guess. Seems like `typesafe.config` is not the simplest library out there)

Reply:
You are right. The same answer as for Sasa:  I have created issue #43 for that and it has to be properly handled.